### PR TITLE
layers: Remove some old TODOs

### PR DESCRIPTION
**Pull request by spencer-lunarg**
cleans out some TODO, comment for each below

### CODE REVIEW

File: layers/core_checks/cc_android.cpp [hunk 1]
```diff
@@ -324,16 +324,6 @@ bool CoreChecks::ValidateAllocateMemoryANDROID(const VkMemoryAllocateInfo *alloc
...
-            // If the Android hardware buffer's usage includes AHARDWAREBUFFER_USAGE_GPU_MIPMAP_COMPLETE, the image must
-            // have either a full mipmap chain or exactly 1 mip level.
-            //
-            // NOTE! The language of this VUID contradicts the language in the spec (1.1.93), which says "The
-            // AHARDWAREBUFFER_USAGE_GPU_MIPMAP_COMPLETE flag does not correspond to a Vulkan image usage or creation flag. Instead,
-            // its presence indicates that the Android hardware buffer contains a complete mipmap chain, and its absence indicates
-            // that the Android hardware buffer contains only a single mip level."
-            //
-            // TODO: This code implements the VUID's meaning, but it seems likely that the spec text is actually correct.
-            // Clarification requested.
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_android.cpp:336]: doing all the AHB stuff, this logic is right, the spec is clear now and I find no reason to keep this as a way to spark confusion

File: layers/core_checks/cc_descriptor.cpp [hunk 2]
```diff
@@ -2528,8 +2528,6 @@ bool CoreChecks::VerifyCopyUpdateContents(const VkCopyDescriptorSet &update, con
                                              "Attempted copy update to sampler descriptor with invalid sampler (%s).",
                                              FormatHandle(update_sampler).c_str());
                         }
-                    } else {
-                        // TODO : Warn here
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_descriptor.cpp:2532]: if things are not immutable samplers, we have checks elsewhere already

File: layers/core_checks/cc_descriptor.cpp [hunk 3]
```diff
@@ -4824,8 +4820,6 @@ bool CoreChecks::PreCallValidateCreatePipelineLayout(VkDevice device, const VkPi
 
     // Extension exposes new properties limits
     if (IsExtEnabled(device_extensions.vk_ext_descriptor_indexing)) {
-        // XXX TODO: replace with correct VU messages
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_descriptor.cpp:4827]: things are correct, seems this was left when VUs were updated

File: layers/core_checks/cc_device_memory.cpp [hunk 4]
```diff
@@ -391,7 +391,6 @@ bool CoreChecks::PreCallValidateAllocateMemory(VkDevice device, const VkMemoryAl
         }
     }
 #endif
-    // TODO: VUIDs ending in 00643, 00644, 00646, 00647, 01745, 00645, 00648, 01744
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_device_memory.cpp:394]: we have proper tracking for this in issues

**Review by arno-lunarg — APPROVED** [review]
LGTM!

**Pull request merged**